Feature/solution upgrade

alastairs opened the pull request:
Make the necessary changes to be able to open GiveCRM in VS2012.  Note that this made some "non-functional" changes to the Web project, to enable backwards compatibility with VS2010, and the Database Projects are now deprecated in favour of SQL Server Data Tools (SSDT).  If you're using VS2010, you'll need to download SSDT from http://go.microsoft.com/fwlink/?LinkID=237127&clcid=0x409.

alastairs committed:
Update to VS2012 RTM.

Note: this also upgrades the Database projects to SQL Server Data Tools
projects.  If you're using VS2010, you'll need to download SSDT from
http://go.microsoft.com/fwlink/?LinkID=237127&clcid=0x409.

alastairs committed:
Update NCrunch files.

Add an NCrunch project for the Business Logic tests assembly, and update
the NCrunch Solution to v1.41.

alastairs committed:
Update Web project.

This commit applies non-functional changes to the Web project to allow it
to open in both VS2010 and VS2012.

alastairs committed:
Minor comment update.

The code was re-generated, which caused a bump in patch number.

alastairs committed:
Update NCrunch solution.

Update the NCrunch solution file for v1.41.

nbarnwell commented:
As with the admin site, if you could just add a note to the readme.md for the benefit of those still running VS2010 I'll go ahead and merge this asap.

alastairs committed:
Update the ReadMe for SSDT.

Include a link to the download for the SQL Server Developer Tools, and
indicate which versions of Visual Studio are supported.

alastairs commented:
Updated the ReadMe with details of SSDT as requested.

alastairs committed:
Fix an error running on .NET 4.5.

The list of Members started failing when run on .NET 4.5, because the
PagedList helper library was causing a VerificationException to be thrown.
This new Exception type is thrown when the code being run is required to
be type safe, but cannot be verified as such.  See
http://msdn.microsoft.com/en-us/library/system.security.verificationexception.aspx
for more information.

Simply upgrading the PagedList helper library to the latest version
resolved the issue.

alastairs committed:
Generate .sql files from the SSDT projects.

alastairs committed:
Instructions for deploying SSDT projects.

Add instructions to the ReadMe describing how to deploy SSDT projects to a
live SQL Server instance (Express or full).
